core/src/banking_stage/qos_service.rs

```diff
@@ -167,17 +168,20 @@ impl QosService {
         bank: &Bank,
     ) {
         let mut cost_tracker = bank.write_cost_tracker().unwrap();
+        let mut num_included = 0;
         transaction_cost_results
             .zip(transaction_committed_status)
             .for_each(|(tx_cost, transaction_committed_details)| {
                 // Only transactions that the qos service included have to be
                 // checked for update
                 if let Ok(tx_cost) = tx_cost {
+                    num_included += 1;
```

Comment:
Want to make sure I understand why the increment is here. I believe it is because:

1. By coming in here, it indicates we have a mix of committed and uncommitted transactions (as opposed to remove_transaction_costs, where everything passed in was not committed)
2. We only add transaction to in flight if it makes it past all the error checks. Thus, we only want to include `tx_cost` that are `Ok` and not `Err`.
3. We don't care about Committed vs. NotCommitted because either way they were in flight and are no longer in flight.

Reply:
1. Exactly
2. Correct. Similar to above, we could have a mix of tx's that fit / did not fit. This is determined by this snippet:
https://github.com/anza-xyz/agave/blob/10d06773cd1daf251aeb7a6283da6bd8d75f571a/core/src/banking_stage/qos_service.rs#L106-L116
3. Almost but not exactly (code snippet further down shows the update case):
    - Transactions that are not committed result in removing the cost completely
    - Transactions that are committed result in updating the cost to use actual instead of estimated
https://github.com/anza-xyz/agave/blob/10d06773cd1daf251aeb7a6283da6bd8d75f571a/core/src/banking_stage/qos_service.rs#L192-L200